record.go

```diff
@@ -38,6 +39,33 @@ func (r *RecordsResponse) FindRecord(id string) (*Record, error) {
 	return nil, notFoundErr
 }
 
+func (r *RecordsResponse) FindRecordByName(name string, wildcard bool) ([]Record, error) {
+	if r.Result == "error" {
+		return nil, fmt.Errorf("API Error: %s", r.Message)
+	}
+
+	objs := r.Response.Recs.Records
+	notFoundErr := errors.New("Record not found")
+
+	// No objects, return nil
+	if len(objs) < 0 {
+		return nil, notFoundErr
+	}
+
+	var recs []Record
+	suffix := "." + name
+
+	for _, v := range objs {
+		if v.Name == name {
+			recs = append(recs, v)
+		} else if wildcard && strings.HasSuffix(v.Name, suffix) {
```

Comment:
Gotcha, of course – early morning comprehension, mistook the `else if` for an `else`